opal/mca/btl/tcp/btl_tcp_endpoint.c

```diff
@@ -728,34 +728,44 @@ static int mca_btl_tcp_endpoint_start_connect(mca_btl_base_endpoint_t* btl_endpo
 
     /* start the connect - will likely fail with EINPROGRESS */
     mca_btl_tcp_proc_tosocks(btl_endpoint->endpoint_addr, &endpoint_addr);
-    
+
     /* Bind the socket to one of the addresses associated with
      * this btl module.  This sets the source IP to one of the 
      * addresses shared in modex, so that the destination rank 
      * can properly pair btl modules, even in cases where Linux 
      * might do something unexpected with routing */
-    opal_socklen_t sockaddr_addrlen = sizeof(struct sockaddr_storage);
+    opal_socklen_t sockaddr_addrlen = sizeof(struct sockaddr);
     if (endpoint_addr.ss_family == AF_INET) {
         assert(NULL != &btl_endpoint->endpoint_btl->tcp_ifaddr);
         if (bind(btl_endpoint->endpoint_sd, (struct sockaddr*) &btl_endpoint->endpoint_btl->tcp_ifaddr,
-		 sockaddr_addrlen) < 0) {
-	    BTL_ERROR(("bind() failed: %s (%d)", strerror(opal_socket_errno), opal_socket_errno));
-
-	    CLOSE_THE_SOCKET(btl_endpoint->endpoint_sd);
-	    return OPAL_ERROR;
-	}
+                 sockaddr_addrlen) < 0) {
+            opal_output_verbose(10, opal_btl_base_framework.framework_output,
+                                "bind on local address (%s:%d) failed: %s (%d)",
+                                opal_net_get_hostname((struct sockaddr*) &btl_endpoint->endpoint_btl->tcp_ifaddr),
+                                htons(((struct sockaddr_in*)&btl_endpoint->endpoint_btl->tcp_ifaddr)->sin_port),
+                                strerror(opal_socket_errno), opal_socket_errno);
+            BTL_ERROR(("bind() failed: %s (%d)", strerror(opal_socket_errno), opal_socket_errno));
```

Comment:
I'm not sure I understand why we'd have two levels of output here.  Seems like better to just always print the full debugging.